week2-section2/exercise2/Spotify.cpp

```diff
@@ -0,0 +1,181 @@
+#include <string>
+
+FSpotify::FSpotify()
+{
+	ActiveMenuCommand = EMenuCommand::MainMenu;
+	PlaylistContainer = FPlaylistContainer(0);
+};
+
+//------------------------------------------------------
+void FSpotify::OpenMainMenu() 
+{
+	std::system("cls");
+	std::cout << "Welcome to Spotify! What would you like to do?\n1 - Create a Playlist\n2 - Browse Playlists\n3 - Exit" << std::endl;
+	int InputConverter;
+	std::cin >> InputConverter;
+	while (std::cin.fail() || (InputConverter <= 0) || (InputConverter >= 4))
+	{
+			std::system("cls");
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			std::cout << "Invalid Option, Insert a valid option" << std::endl;
+			std::cout << "Welcome to Spotify! What would you like to do?\n1 - Create a Playlist\n2 - Browse Playlists\n3 - Exit" << std::endl;
+			std::cin >> InputConverter;
+	}
+
+	ActiveMenuCommand = (EMenuCommand)InputConverter;
+	std::system("cls");
+	FSpotify::RunApp();
+};
+
+//--------------------------------------------------------
+void  FSpotify::OpenCreatePlaylistMenu() 
+{
+	//Creating the playlist
+	std::system("cls");
+	std::cout << "Please enter a name for the playlist: " << std::endl;
+	char TitlePlaylist[500];
+	std::cin.getline(TitlePlaylist, 500,'\n');
+	std::cin >> TitlePlaylist;
+	FPlaylist Playlist = FPlaylist(TitlePlaylist, 0);
+	
+	std::cout << "Let's add a song to " << TitlePlaylist << std::endl;
+	//Creating the song loop
+
+	bool bWantToAddASong = true;
+	while (bWantToAddASong == true)
+	{
+		std::cout << "Please enter the song title: " << std::endl;
+		char TitleSong[500];
+		std::cin.getline(TitleSong, 500, '\n');
+		std::cin >> TitleSong;
+
+		std::cout << "Please enter the artist name: " << std::endl;
+		char ArtistName[500];
+		std::cin.getline(ArtistName, 500, '\n');
+		std::cin >> ArtistName;
+
+		std::cout << "Please enter the duration (in seconds) of the song: " << std::endl;
+		int DurationSong;
+		std::cin >> DurationSong;
+		while (std::cin.fail())
+		{
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			std::cout << "Invalid Option, insert the duration in seconds " << std::endl;
+			std::cin >> DurationSong;
+		}
+		
+		FSong Song = FSong(TitleSong, ArtistName, DurationSong);
+		
+		//adding the song in playlist
+		Playlist.AddSong(Song);
+
+		//menu
+		std::cout << "1 - Enter another song\n2 - Go back to main menu" << std::endl;
+		int MenuSelection;
+		std::cin >> MenuSelection;
+		while (std::cin.fail() || (MenuSelection < 1) || (MenuSelection > 2))
+		{
+			std::system("cls");
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			std::cout << "Invalid Option, Insert a valid option" << std::endl;
+			std::cout << "1 - Enter another song\n2 - Go back to main menu" << std::endl;
+			std::cin >> MenuSelection;
+		}
+		if (MenuSelection == 2)
+		{
+			//adding the playlist in container
+			PlaylistContainer.AddPlaylist(Playlist);
+			bWantToAddASong = false;
+			FSpotify::OpenMainMenu();
+		}
+		else if (MenuSelection == 1) {
```

Comment:
This might be a small comment, but I find it a little misleading to have the `MenuSelection == 2` as the first condition and `MenuSelection == 1` as the second condition. I would've expected it to be the other way around to keep the number order, what do you think?